photutils/psf/tests/test_daogroup.py

```diff
@@ -0,0 +1,206 @@
+         0.5 +                                                         +
+             |                                                         |
+             |                                                         |
+           0 +  *   *   *    *   *                 *   *   *    *   *  +
+             |                                                         |
+             |                                                         |
+        -0.5 +                                                         +
+             |                                                         |
+             |                                                         |
+             |                                                         |
+          -1 +--+-------+--------+--------+--------+-------+--------+--+
+                0      0.5       1       1.5       2      2.5       3
+        """
+
+        first_group = Table([np.linspace(0, 1, 5), np.zeros(5), np.arange(5)],
+                            names=('x_0', 'y_0', 'id'))
+        second_group = Table([np.linspace(2, 3, 5), np.zeros(5),
+                              5 + np.arange(5)], names=('x_0', 'y_0', 'id'))
+        starlist = vstack([first_group, second_group])
+        true_groups = [first_group, second_group]
+        daogroup_groups = daogroup(starlist, crit_separation=0.3)
+        assert_array_equal(true_groups, daogroup_groups)
+
+    def test_daogroup_four(self):
+        """
+             +-+---------+---------+---------+---------+-+
+           1 +                     *                     +
+             |           *                   *           |
+             |                                           |
+             |                                           |
+         0.5 +                                           +
+             |                                           |
+             |                                           |
+             |                                           |
+           0 + *                                       * +
+             |                                           |
+             |                                           |
+        -0.5 +                                           +
+             |                                           |
+             |                                           |
+             |           *                   *           |
+          -1 +                     *                     +
+             +-+---------+---------+---------+---------+-+
+              -1       -0.5        0        0.5        1
+        """
+
+        x = np.linspace(-1., 1., 5)
+        y = np.sqrt(1. - x**2)
+        xx = np.hstack((x, x))
+        yy = np.hstack((y, -y))
+        starlist = Table([xx, yy, np.arange(10)], names=('x_0', 'y_0', 'id'))
+        daogroup_group = daogroup(starlist, crit_separation=2.5)
+        assert_array_equal([starlist], daogroup_group)
+
+    def test_daogroup_five(self):
```

Comment:
``np.full`` was new in numpy 1.8, so you can import ``NUMPY_LT_1_8``, and mark the test xfail or skip if it's true.

e.g.
``from astropy.utils.compat import NUMPY_LT_1_8`` and decorate this test with
 ``@pytest.mark.xfail(NUMPY_LT_1_8, reason="Numpy 1.8 or later is required")``